<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SimulationFrame</class>
 <widget class="QWidget" name="SimulationFrame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>268</width>
    <height>557</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="whatsThis">
   <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;The simulation frame lists all tasks of the current simulation setup.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;New tasks can be created, their parameters can be changed, and tasks can be started stopped or paused individually or all together.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>0</number>
     </property>
     <property name="midLineWidth">
      <number>2</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QFrame" name="tasksFrame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>4</number>
       </property>
       <property name="bottomMargin">
        <number>9</number>
       </property>
       <item>
        <widget class="QWidget" name="headerWidget" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <pointsize>8</pointsize>
          </font>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <property name="verticalSpacing">
           <number>0</number>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="headerLabel">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Simulation:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QWidget" name="statusWidget" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="statusLabel">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Initialized</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>74</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
            <zorder>horizontalSpacer2</zorder>
            <zorder>statusLabel</zorder>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="tasksControlWidget" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <property name="spacing">
           <number>4</number>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="tasksControlTopLayout">
            <property name="spacing">
             <number>4</number>
            </property>
            <item>
             <widget class="QComboBox" name="tasksComboBox">
              <property name="toolTip">
               <string>List of task types. Choose one and click the button to add another task.</string>
              </property>
              <property name="whatsThis">
               <string>Chooses a task type to add.</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tasksAddButton">
              <property name="toolTip">
               <string>Adds another task of the currently selected type.</string>
              </property>
              <property name="whatsThis">
               <string>Add a new task of the type chosen in the selection box.
There may only be one server task at a time.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../carbon/imageresources.qrc">
                <normaloff>:/add</normaloff>:/add</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="tasksControlBottomLayout">
            <property name="spacing">
             <number>3</number>
            </property>
            <item>
             <widget class="QToolButton" name="tasksPlayButton">
              <property name="toolTip">
               <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Starts the simulation and runs all tasks.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;All tasks are started at the same time.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="whatsThis">
               <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Starts the simulation and runs all tasks.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Warning: All tasks are started at the same time. Some processes or tasks may need to be started one after another. Start them one by one to prevent crashes.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../carbon/imageresources.qrc">
                <normaloff>:/play</normaloff>:/play</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tasksPauseButton">
              <property name="toolTip">
               <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Pauses the simulation by pausing all tasks.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;All tasks are paused at the same time.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="whatsThis">
               <string>Pauses the simulation by pausing all tasks.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../carbon/imageresources.qrc">
                <normaloff>:/pause</normaloff>:/pause</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tasksStopButton">
              <property name="toolTip">
               <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Stops the simulation by resetting all SimSpark entities and stopping the simulation tasks.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;All tasks are stopped at &lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;the same time.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="whatsThis">
               <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Stops the simulation by resetting all SimSpark entities and stopping the tasks.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Warning: All tasks are started at the same time. Some processes or tasks may need to be stopped one after another. Stop them one by one to prevent crashes.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../carbon/imageresources.qrc">
                <normaloff>:/stop</normaloff>:/stop</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tasksRestartButton">
              <property name="toolTip">
               <string>Restarts the simulation by stopping and running all tasks.</string>
              </property>
              <property name="whatsThis">
               <string>Restarts the simulation by stopping and running all tasks.</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../carbon/imageresources.qrc">
                <normaloff>:/previous</normaloff>:/previous</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="Line" name="line">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tasksExpandButton">
              <property name="toolTip">
               <string>Expands detailed information of all tasks.</string>
              </property>
              <property name="whatsThis">
               <string>Expands detailed information of all tasks.</string>
              </property>
              <property name="text">
               <string extracomment="Show details for all tasks."/>
              </property>
              <property name="icon">
               <iconset resource="../../carbon/imageresources.qrc">
                <normaloff>:/arrow_expand</normaloff>:/arrow_expand</iconset>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tasksCollapseButton">
              <property name="toolTip">
               <string>Collapses detailed information of all tasks.</string>
              </property>
              <property name="whatsThis">
               <string>Collapses detailed information of all tasks.</string>
              </property>
              <property name="text">
               <string extracomment="Show only header for all tasks."/>
              </property>
              <property name="icon">
               <iconset resource="../../carbon/imageresources.qrc">
                <normaloff>:/arrow_collapse</normaloff>:/arrow_collapse</iconset>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QScrollArea" name="tasksScrollArea">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="tasksScrollAreaContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>248</width>
            <height>457</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <widget class="QWidget" name="tasksWidget" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout"/>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>2000</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../carbon/imageresources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>tasksComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SimulationFrame</receiver>
   <slot>chooseTaskType(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>241</x>
     <y>31</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tasksAddButton</sender>
   <signal>clicked()</signal>
   <receiver>SimulationFrame</receiver>
   <slot>addTask()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>63</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tasksPlayButton</sender>
   <signal>clicked()</signal>
   <receiver>SimulationFrame</receiver>
   <slot>runTasks()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>27</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>-6</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tasksPauseButton</sender>
   <signal>clicked()</signal>
   <receiver>SimulationFrame</receiver>
   <slot>pauseTasks()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>-2</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tasksStopButton</sender>
   <signal>clicked()</signal>
   <receiver>SimulationFrame</receiver>
   <slot>stopTasks()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>80</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>-2</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tasksRestartButton</sender>
   <signal>clicked()</signal>
   <receiver>SimulationFrame</receiver>
   <slot>restartTasks()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>-6</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tasksExpandButton</sender>
   <signal>clicked()</signal>
   <receiver>SimulationFrame</receiver>
   <slot>expandTasks()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>240</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tasksCollapseButton</sender>
   <signal>clicked()</signal>
   <receiver>SimulationFrame</receiver>
   <slot>collapseTasks()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>chooseTaskType(int)</slot>
  <slot>addTask()</slot>
  <slot>runTasks()</slot>
  <slot>pauseTasks()</slot>
  <slot>stopTasks()</slot>
  <slot>restartTasks()</slot>
  <slot>expandTasks()</slot>
  <slot>collapseTasks()</slot>
 </slots>
</ui>
